libs/database/src/lib/entities/delayed-job.entity.ts

```diff
@@ -44,5 +62,56 @@ export class DelayedJob extends Model<DelayedJob> {
 
   @Column(BOOLEAN)
   isAcknowledged: boolean;
-  
+
+  @AllowNull
+  @Column(STRING)
+  name: string | null;
+
+  @AllowNull
+  @Column(STRING)
+  entityType: string | null;
+
+  @ForeignKey(() => Project)
+  @ForeignKey(() => Site)
+  @Column
+  entityId: number;
+
+  @BelongsTo(() => Project, {
+    foreignKey: "entityId",
+    constraints: false,
+    scope: {
+      entityType: "App\\Models\\V2\\Projects\\Project"
+    },
+    as: "entityProject"
+  })
+  entityProject?: Project;
+
+  @BelongsTo(() => Site, {
+    foreignKey: "entityId",
+    constraints: false,
+    scope: {
+      entityType: "App\\Models\\V2\\Sites\\Site"
+    },
+    as: "entitySite"
+  })
+  entitySite?: Site;
+
+  declare getEntityProject: () => Promise<Project | null>;
+  declare getEntitySite: () => Promise<Site | null>;
+
+  async getRelatedEntity(): Promise<string | null> {
+    if (!this.entityId) return null;
+
+    if (this.entityType === "App\\Models\\V2\\Projects\\Project") {
+      const project = this.entityProject ?? (await this.getEntityProject());
+      return project?.name ?? null;
```

Comment:
If you're needing to load the relation, it's best to cache it on this instance so that a further process in this thread doesn't have to do it again. I would set these lines as 
```
this.entityProject ??= await this.$get('entityProject');
return this.entityProject?.name ?? null:
```

Reply:
changed, nonetheless couldnt use you code exactly because there was an error that i couldnt fix it.
Even though is good.

Reply:
Hmm, maybe the `??=` operator isn't set up in this codebase? If that's not working, then this code should still do it the more verbose way:
```
if (this.entityProject == null) {
  this.entityProject = await this.$get("entityProject");
}
return this.entityProject?.name;
```